framework/src/materials/FunctorADConverter.C

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorADConverter.h"
+
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("MooseApp", FunctorADConverter);
+registerMooseObject("MooseApp", VectorFunctorADConverter);
+
+template <typename T>
+InputParameters
+FunctorADConverterTempl<T>::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription("Converts regular functors to AD functors and "
+                             " AD functors to regular functors");
+  params.addParam<std::vector<std::string>>(
+      "reg_props_in", "The names of the regular functors to convert to AD functors");
+  params.addParam<std::vector<std::string>>("ad_props_out", "The names of the output AD functors");
+  params.addParam<std::vector<std::string>>(
+      "ad_props_in", "The names of the AD functors to convert to regular functors");
+  params.addParam<std::vector<std::string>>("reg_props_out",
+                                            "The names of the output regular functors");
+  return params;
+}
+
+template <typename T>
+FunctorADConverterTempl<T>::FunctorADConverterTempl(const InputParameters & parameters)
+  : FunctorMaterial(parameters)
+{
+  auto reg_props_in = getParam<std::vector<std::string>>("reg_props_in");
```

Comment:
It'd be more elegant to use `getParam<std::string, std::string>("reg_props_in", "ad_props_out");` to have MOOSE build a vector of pairs

Reply:
could do a map too, but all the Generic Materials are not built this way